Source/Core/Core/PowerPC/GDBThread.h

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// Originally written by Sven Peter <sven@fail0verflow.com> for anergistic.
+
+#pragma once
+
+#include <signal.h>
+
+#include "Common/CommonTypes.h"
+#include "Common/Thread.h"
+#include "Common/Event.h"
+
+#include "Core/HW/CPU.h"
+#include "Core/HW/Memmap.h"
+#include "Core/PowerPC/PowerPC.h"
+
+#ifdef _WIN32
+#include <ws2tcpip.h>
+#include <iphlpapi.h>
+#include <iphlpapi.h>
+#else
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#endif
+#include <stdarg.h>
+
+#define		GDB_BFR_MAX	10000
+
+// --------------------------------------------------------------------------------------
+//  GDBThread
+// --------------------------------------------------------------------------------------
+// Threaded wrapper class for implementing debugging functionality through a gdb stub.
+//
+class GDBThread
+{
+public:
+    GDBThread();
+    ~GDBThread();
+
+    bool Initialize();
+    void Terminate();
+
+protected:
+    void OnStart();
+    void OnStop();
+    void OnPause();
+    void OnResume();
+    void UpdateState(CPU::State current_state);
+
+    void ExecuteTaskInThread();
```

Comment:
Since this is the function that runs on the separate thread, I think this name doesn't really fit.